src/figuresClasses.ts

```diff
@@ -1,11 +1,91 @@
-export interface Figure {}
+/* eslint-disable @typescript-eslint/lines-between-class-members */
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: string;
+  color: 'red' | 'green' | 'blue';
 
-export class Circle implements Figure {}
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape = 'triangle';
+  color: 'red' | 'green' | 'blue';
+  private a: number;
+  private b: number;
+  private c: number;
+
+  constructor(
+    color: 'red' | 'green' | 'blue',
+    a: number,
+    b: number,
+    c: number,
+  ) {
+    this.color = color;
+
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Side lengths must be greater than 0');
+    }
+
+    const sides = [a, b, c].sort((x, z) => z - x);
+
+    if (sides[0] >= sides[1] + sides[2]) {
+      throw new Error('This triangle cannot exist');
+    }
```

Comment:
sort???
u can use `Math.max` method for this logic
```suggestion
    const maxNum = Math.max(a, b, c); // max size
    const sum = a + b + c; // all the sides

    if (maxNum >= sum - maxNum) { // invalid triangle
      throw new Error('This triangle cannot exist');
    }
```